test/spec/modules/kargoBidAdapter_spec.js

```diff
@@ -1050,148 +1785,205 @@ describe('kargo adapter tests', function () {
+        { applicableSections: {}, consentString: 'test-consent-string', as: '', cs: 'test-consent-string' },
+        { applicableSections: [], consentString: 'test-consent-string', as: '', cs: 'test-consent-string' },
+        { as: '', cs: '' },
+      ].forEach(value => expect(getUserSyncs(undefined, undefined, value), `Value - ${value}`)
+        .to.deep.equal(buildSyncUrls(baseUrl
+          .replace(/gpp=/, `gpp=${value.cs}`)
+          .replace(/gpp_sid=/, `gpp_sid=${value.as}`))));
     });
 
-    it('pass through gpp consent', function () {
-      turnOnClientId();
-      safelyRun(() => expect(getUserSyncsWhenAllowed(null, null, { consentString: 'gppString', applicableSections: [-1] })).to.deep.equal(getSyncUrls('', '', '', 'gppString', '-1')));
+    it('handles malformed gpp information', function() {
+      [
+        null,
+        false,
+        true,
+        1,
+        '1',
+        'test-applies',
+        [],
+        {}
+      ].forEach(value => expect(getUserSyncs(undefined, undefined, value), `Value - ${JSON.stringify(value)}`)
+        .to.deep.equal(buildSyncUrls(baseUrl)));
     });
 
-    it('no user syncs when there is outdated browser', function () {
-      turnOnClientId();
-      simulateOutdatedBrowser();
-      safelyRun(() => expect(getUserSyncsWhenAllowed()).to.be.an('array').that.is.empty);
+    it('includes all 3 if provided', function() {
+      expect(getUserSyncs(
+        { gdprApplies: true, consentString: 'test-gdpr-consent' },
+        '1---',
+        { applicableSections: [1, 2, 3], consentString: 'test-gpp-consent' }
+      )).to.deep.equal(buildSyncUrls(baseUrl
+        .replace(/gdpr=\d/, 'gdpr=1')
+        .replace(/gdpr_consent=/, 'gdpr_consent=test-gdpr-consent')
+        .replace(/us_privacy=/, 'us_privacy=1---')
+        .replace(/gpp=/, 'gpp=test-gpp-consent')
+        .replace(/gpp_sid=/, 'gpp_sid=1,2,3')));
     });
+  });
 
-    it('no user syncs when no iframe syncing allowed', function () {
-      turnOnClientId();
-      safelyRun(() => expect(getUserSyncsWhenForbidden()).to.be.an('array').that.is.empty);
+  describe('supportedMediaTypes', function() {
+    it('exposes video and banner', function() {
+      expect(spec.supportedMediaTypes).to.deep.equal([ 'banner', 'video' ]);
     });
   });
 
-  describe('timeout pixel trigger', function () {
-    let triggerPixelStub;
+  describe('onTimeout', function() {
+    before(function() {
+      sinon.spy(utils, 'triggerPixel');
+    });
+
+    beforeEach(function() {
+      utils.triggerPixel.resetHistory();
+    });
 
-    beforeEach(function () {
-      triggerPixelStub = sinon.stub(utils, 'triggerPixel');
+    it('does not call triggerPixel if timeout data is not provided', function() {
+      spec.onTimeout(null);
+      expect(utils.triggerPixel.callCount).to.equal(0);
     });
 
-    afterEach(function () {
-      utils.triggerPixel.restore();
+    it('calls triggerPixel if any timeout data is provided', function() {
```

Comment:
Prob don't need two of these, right?
https://github.com/KargoGlobal/Prebid.js/pull/31/files#diff-2fe18725d9d2620649c2894853c6a5f75ca201cad0b65ebdfa37ac37d48e676eR1976

Reply:
Good callout, I will remove the first test